Cython/Compiler/ExprNodes.py

```diff
@@ -13356,6 +13356,31 @@ def analyse_cpp_comparison(self, env):
         type1 = self.operand1.type
         type2 = self.operand2.type
         self.is_pycmp = False
+
+        if self.operator in ['in', 'not_in']:
+            begin = type2.scope.lookup("begin")
+            end = type2.scope.lookup("end")
+            if begin and end:
+                self.exception_check = '+'
+                self.exception_value = None
+                if needs_cpp_exception_conversion(self):
```

Comment:
I think we should read the exception configuration from the C++ methods, shouldn't we? And neither `begin()`, `end()`, `find()`, nor `contains()` are declared as raising exceptions at least in our libcpp STL declarations. However, we also have to deal with user implemented containers here, so I'd suggest looking up all four methods and trying to find the most appropriate exception setup from that, if at least one of them declares exceptions. I'd go for `contains()` first, then look up `find()`, then fall back to `end()`, then `begin()`, and take the first exception configuration that we find.

Note that, if we need to handle exceptions, then we also need to set `self.is_temp = True` to make sure that we pass the return value through a temporary variable inside of the try-block.